torch_geometric/transforms/add_positional_encoding.py

```diff
@@ -0,0 +1,151 @@
+from typing import Any, Optional
+
+import numpy as np
+import torch
+from scipy.sparse.linalg import eigs, eigsh
+from torch import Tensor
+from torch_scatter import scatter_add
+from torch_sparse import spspmm
+
+from torch_geometric.data import Data
+from torch_geometric.data.datapipes import functional_transform
+from torch_geometric.transforms import BaseTransform
+from torch_geometric.utils import (
+    get_laplacian,
+    remove_self_loops,
+    to_scipy_sparse_matrix,
+)
+from torch_geometric.utils.num_nodes import maybe_num_nodes
+
+
+def diagonal_weight(edge_index, edge_weight, num_nodes=None):
```

Comment:
Worth adding to `torch_geometric.utils.loop`?

Reply:
That is a good idea. I will work on it first.